### PR TITLE
`Quiz exercises`: Modernize the chart visualization for the quiz point distribution

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

The diagram for the point distribution of a quiz is currently done via chart.js. This PR replaces the chart.js parts with ngx-charts.

### Description

The diagram should be near to the old layout, but some differences are noticeable. Tell me what you think about it! 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Quiz exercise with participants that ideally have a noticeable point distribution
- **Important:** before deploying this change to a test server, go take a look at how the point distribution chart of the quiz looks so far (maybe take a screenshot in order to compare later on). 

Remark: I don't know why the axis labels are flipped currently, but I think this way it makes more sense.

1. Log in to Artemis
2. Navigate to Course Administration
3. Got to the exercises tab of the course where the quiz is already prepared (you can take [this chart](https://vmbruegge60.in.tum.de/course-management/2164/quiz-exercises/15317/quiz-point-statistic) on TS3 to save some time)
4. Click on Quiz Statistics
5. Inspect the presented chart

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

**Old chart:**
image

**New chart:**
image

### CODE REVIEW

**Review by julian-christl — DISMISSED** [review]
Please try to remove the rotation of the labels.
Could `rotateXAxisTicks` do the trick?
https://swimlane.gitbook.io/ngx-charts/examples/bar-charts/vertical-bar-chart

**Review by FlorianGlombik — DISMISSED** [review]
I think that the numbers should not be diagonal  _(on first sight I thought the "3" would be an "rn")_ and the percentage should be readable on first sight _(so removing the on hover and directly displaying it)_.
image

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
         "@ngx-translate/http-loader": "6.0.0",
         "@popperjs/core": "2.10.2",
         "@sentry/browser": "6.13.3",
+        "@swimlane/ngx-charts": "19.1.0",
```
Comment (Olimbia23) [inline package.json:32]: `ngx-charts` is already defined on line 58

**Review by Olimbia23 — APPROVED** [review]
Code changes look good.
Nice work @willergeorg
Reapprove 👍

File: package.json (hunk 1 above)
Comment (willergeorg) [reply package.json:32]: Strange that it is duplicated. I'll remove it

Comment (willergeorg) [reply package.json]: removed in fef44a9

**Review by julian-christl — APPROVED** [review]
Code changes look good.
The screenshot is up to date and reflects a good implementation

File: src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts [hunk 2]
```diff
@@ -115,13 +115,17 @@ export interface DataSetProvider {
      */
     loadDataInDiagram(): void;
 }
-
+// wrapper function to ensure that other ng2-implementations still work; should be removed as soon as not used anymore
```
Comment (FlorianGlombik) [inline src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:118]: I would prefer this to be a real documentation comment, note that the argument needs to be covered aswell, so this suggestion should not be committed directly but should just give an idea of what I mean.
```suggestion
/**
* wrapper function to ensure that other ng2-implementations still work; should be removed as soon as not used anymore
*/
```

**Review by FlorianGlombik — DISMISSED** [review]
The table looks fine to me!
image

(_It might look a little bit better if the counts and percentages were bought centralized and might make the table more robust for small screen sizes, but it can also be left as it is, as the counts would not be the only 'issue' on small screen sizes - it is anyways still readable and does not break before 375x812 px - so I do not think that this is an issue that needs much time to be invested into it._)
image

File: src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts (hunk 2 above)
Comment (willergeorg) [reply src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts]: I changed the comment in d0f2078

**Review by FlorianGlombik — APPROVED** [review]
Re-approve, only documentation changes since last review

**Pull request merged**